src/snapshot_middleware/mod.rs

```diff
@@ -388,3 +392,58 @@ pub fn default_sync_rules() -> &'static [SyncRule] {
         ]
     })
 }
+
+fn filter_default_property(
+    snapshot: &SyncbackSnapshot,
+    new_inst: &Instance,
+    name: &str,
+    value: &Variant,
+    attributes: &mut BTreeMap<String, UnresolvedValue>,
+    properties: &mut BTreeMap<String, UnresolvedValue>,
+) {
+    let db = rbx_reflection_database::get();
+    let class_descriptor = db.classes.get(new_inst.class.as_str());
+
+    match value {
+        Variant::Attributes(attrs) => {
+            for (attr_name, attr_value) in attrs.iter() {
+                // We (probably) don't want to preserve internal attributes,
+                // only user defined ones.
+                if attr_name.starts_with("RBX") {
+                    continue;
+                }
+                attributes.insert(
+                    attr_name.clone(),
+                    UnresolvedValue::from_variant_unambiguous(attr_value.clone()),
+                );
+            }
+        }
+        Variant::SharedString(_) => {
+            log::warn!(
+                "Rojo cannot serialize the property {}.{name} in JSON files.\n\
+                 If this is not acceptable, resave the Instance at '{}' manually as an RBXM or RBXMX.",
+                new_inst.class, snapshot.get_new_inst_path(new_inst.referent())
+            )
+        }
+        _ => {
+            let new_prop_is_default = if let Some(class_descriptor) = class_descriptor {
+                if let Some(default) = db.find_default_property(class_descriptor, name) {
+                    variant_eq(value, default)
+                } else {
+                    false
+                }
+            } else {
+                false
+            };
+
+            if new_prop_is_default {
+                properties.remove(name);
```

Comment:
This only really does anything for the project middleware, which might be confusing

Reply:
It's for sure worth adding a comment here. Right now it's not obvious that `properties` might already contain the value we're inserting, so this just looks redundant.